File: src/test/java/com/boot/PetControllerTest.java
package com.boot;

import com.boot.model.PetEntity;
import com.boot.model.PetRepository;
import org.junit.Before;
import org.junit.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import com.boot.controller.PetController;

public class PetControllerTest {
	@InjectMocks
	private PetController petController;

	@Mock
	private PetRepository petRepository;

    @Before
    public void init() {
        MockitoAnnotations.initMocks(this);
    }

	@Test
	public void testShipwreckGet() {
    	PetEntity pet = new PetEntity();
        pet.setId(1l);
//		when(petRepository.findById(1l)).thenReturn(pet);
//
//		PetEntity wreck = petController.get(1L);
//
//		verify(petRepository).findById(1l);
//
//		assertEquals(1l, pet.getId().longValue());
//		assertThat(pet.getId(), is(1l));
	}
}

File: src/test/java/com/boot/PetControllerTest.kt
package com.boot

import com.boot.controller.BaseController
import com.boot.model.PetEntity
import com.boot.service.PetService
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito

import com.boot.controller.PetController
import org.mockito.junit.MockitoJUnitRunner

import org.junit.Assert
import org.mockito.MockitoAnnotations.initMocks
import javax.persistence.EntityNotFoundException

@RunWith(MockitoJUnitRunner::class)
class PetControllerTest: BaseController() {
    @InjectMocks
    lateinit var petController: PetController

    @Mock
    var petService = PetService()

    private val petEntity = PetEntity(
        name = "1",
        gender = "male",
        owner = null
    )

    @Before
    fun init() {
        initMocks(this)
    }

    @Test
    fun shouldReturnPetResponse() {
        var petResponse = petEntity modelOf ""
        Mockito.`when`(petService.getRecord(1)).thenReturn(petResponse)

        var result = petController.getRecord(1)

        Assert.assertNotNull(result)
        Assert.assertEquals(result, getResponse(petResponse))
    }

    @Test(expected = EntityNotFoundException::class)
    fun shouldThrowException() {
        var recordId = 1L
        var petException = EntityNotFoundException("There is no entity exists with id $recordId !")

        Mockito.`when`(petService.getRecord(recordId)).thenThrow(petException)
        var result = petController.getRecord(1)
    }
}
